store/CSDK/test/main.cpp

```diff
@@ -740,6 +740,130 @@ bool testWriteData(JNIEnv *env, char *path, int argc, char *argv[]) {
+            jobject stringField = env->NewStringUTF(a);
+            env->SetObjectArrayElement(arr, 1, stringField);
+
+
+            jobject string2Field = env->NewStringUTF(a);
+            env->SetObjectArrayElement(arr, 2, string2Field);
+
+
+            jobject timeField = env->NewStringUTF("2019-02-12 03:03:34");
+            env->SetObjectArrayElement(arr, 3, timeField);
+
+
+            jobject int4Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 4, int4Field);
+
+            jobject string5Field = env->NewStringUTF(a);
+            env->SetObjectArrayElement(arr, 5, string5Field);
+
+            jobject int6Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 6, int6Field);
+
+            jobject string7Field = env->NewStringUTF(a);
+            env->SetObjectArrayElement(arr, 7, string7Field);
+
+            jobject int8Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 8, int8Field);
+
+            jobject time9Field = env->NewStringUTF("2019-02-12 03:03:34");
+            env->SetObjectArrayElement(arr, 9, time9Field);
+
+            jobject dateField = env->NewStringUTF(" 2019-03-02");
+            env->SetObjectArrayElement(arr, 10, dateField);
+
+            jobject int11Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 11, int11Field);
+
+            jobject int12Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 12, int12Field);
+
+            jobject int13Field = env->NewStringUTF(ctrInt);
+            env->SetObjectArrayElement(arr, 13, int13Field);
+
+            writer.write(arr);
+
+            env->DeleteLocalRef(stringField);
+            env->DeleteLocalRef(string2Field);
+            env->DeleteLocalRef(intField);
+            env->DeleteLocalRef(int4Field);
+            env->DeleteLocalRef(string5Field);
+            env->DeleteLocalRef(int6Field);
+            env->DeleteLocalRef(dateField);
+            env->DeleteLocalRef(timeField);
+            env->DeleteLocalRef(string7Field);
+            env->DeleteLocalRef(int8Field);
+            env->DeleteLocalRef(int11Field);
+            env->DeleteLocalRef(int12Field);
+            env->DeleteLocalRef(int13Field);
+            env->DeleteLocalRef(arr);
+        }
+        writer.close();
+
+        CarbonReader carbonReader;
+        carbonReader.builder(env, path);
+        carbonReader.build();
+        int i = 0;
+        int printNum = 10;
+        CarbonRow carbonRow(env);
+        while (carbonReader.hasNext()) {
+            jobject row = carbonReader.readNextRow();
+            i++;
+            carbonRow.setCarbonRow(row);
+            if (i < printNum) {
+                printf("%s\t%d\t%ld\t", carbonRow.getString(1));
+            }
+            env->DeleteLocalRef(row);
+        }
+        carbonReader.close();
+    } catch (jthrowable ex) {
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+    }
```

Comment:
why not use `__finally` block to close the reader

Reply:
done